components/brave_wallet/resources/brave_wallet_provider.js

```diff
@@ -17,4 +17,31 @@
       BraveWeb3ProviderEventEmitter.removeAllListeners
   // For webcompat
   window.ethereum.isMetaMask = true
+  var alreadyLogged = false
+  var logweb3Warning = () => {
+    if (!alreadyLogged) {
+      console.warn('You are accessing the window.web3 shim. This object is deprecated, please use window.ethereum instead.')
+      alreadyLogged = true
```

Comment:
actually do we have to suppress the warning? I think it has to be shown every time this shim is used as an aggressive reminder.

Reply:
Per Slack. I'll leave it as it is so we do not spam the console